### ISSUE TITLE
OutputNode is not working correctly in some case

### ISSUE THREAD

**Issue by MaeThird** [ARM-software/ComputeLibrary#455]
I'm talking on the assumptions that the InputNode,ConstNode and OutputNode are torching data through interface  `bool access_tensor(ITensor &tensor) override` in `class graph::ITensorAccessor`. That's **the InputNode is fetching image data,the ConstNode('Mean','Variance','Beta'...) is fetching trained weights and the OutputNode is pushing out the calculated data**.
There would be an issue when **a OutputNode is following an ActivationLayer whose
ActivationFunction is NOT LEAY_RELU**(this is anther bug explained) **and is following a  BatchNormalizationLayer**. 
Such as these piece code:
```
... 
ConvolutionLayer conv_0(some params );
  BatchNormalizationLayer bn_0(mean,variance,gama,beta params);
  ActivationLayer ac_0(ActivationLayerInfo(ActivationLayerInfo::ActivationFunction::RELU));
  OutputLayer branch_out_0(some params);

  graph.add_layer(conv_0);
  graph.add_layer(bn_0);
  graph.add_layer(ac_0);
  graph.add_layer(branch_out_0);

  GraphConfig cfg;
  cfg.use_tuner = 0;
  cfg.use_transition_memory_manager = false;
  graph.finalize(Target::NEON,cfg);
```

The  **access_tensor bound to  branch_out_0 won't be called**.
**According to the work flow the reason is** : After the NodeFusion processing the ac_0 layer was merged into bn_0 layer and ac_0 layer was removed although the input_edge to the following branch_out_0 had been updated through `Graph::add_connection` but **the accessor did not had been updated**(actually it was only set once through `Graph:: set_accessor_on_node `when it was initially added to graph)  resulting the call to `call_accesor()` in graph/Tensor.cpp returned ahead.

The BatchnormalizationNode would have some ConstNodes and each of them has its' own data accessor so it's not intuitive to set the OutputNode's accessor to thess Tensors.

**Comment by GeorgeARM** [comment]
Hello @MaeThird,

We are aware of this bug. For now you can just disable the node fusion pass and it should be fine.
We just have have to move the accessor as its bound to the tensor that is getting removed when performing the fusion. Thanks for raising.

**Comment by MaeThird** [comment]
I'm not quite sure if it is caused by node fusion.I'm only sure is the Tensor that the OutputNode's input bounded to was "removed" by some reason.
Here are 3 pics I drew. **First one is before the call to** `graph.finalize(Target::NEON,cfg);`
![t1](https://user-images.githubusercontent.com/14851411/40873351-2a588464-664e-11e8-8137-595093e51b6d.png)
**Second one is commented the**   `pm.append(support::cpp14::make_unique<NodeFusionMutator>());`
![t2](https://user-images.githubusercontent.com/14851411/40873358-52998dce-664e-11e8-920e-ee74cc68292a.png)
**Third one is with node fusion open:**
![t3](https://user-images.githubusercontent.com/14851411/40873368-77bc5050-664e-11e8-88c4-31ed1c6cc0f9.png)
The "T" is the Tensor used by Node."Node_9" is the ac_0."Node_10" is the branch_out_0 logically but not physically(physically it has only 'input' data stored in Tensor3).
The last two pics showed whether node fusion is open or not the actual tensor storing the data is Tensor3. The Tensor8 and Tensor9 is optimized for any reason.
I hope it will be help.

**Comment by GeorgeARM** [comment]
Hmmm I see, thus might be the InPlace mutator. Eitherway the problem is that during the mutations we accidentally remove the accessor that is bound to the output node. Try commenting the inplace mutator.
